### PR TITLE
Add rtl direction

### PR DESCRIPTION
**Pull request by saeedrahimi**
Just added some fixes for rtl direction.
Fixed columns in rtl direction will be displayed correctly with these changes.

### CODE REVIEW

**Comment by zombieJ** [comment]
Could you help to add a test case to test related class?

**Comment by saeedrahimi** [comment]
> Could you help to add a test case to test related class?

I've added some tests. ;)

File: src/hooks/useColumns.tsx [hunk 1]
```diff
@@ -167,17 +204,17 @@ function useColumns<RecordType>(
...
-
     return finalColumns;
-  }, [transformColumns, withExpandColumns]);
-
-  const flattenColumns = React.useMemo(() => flatColumns(mergedColumns), [mergedColumns]);
+  }, [transformColumns, withExpandColumns, direction]);
 
+  const flattenColumns = React.useMemo(
+    () => reverseFixedForRtlDirection(flatColumns(mergedColumns, direction), direction),
+    [mergedColumns],
```
Comment (zombieJ) [inline src/hooks/useColumns.tsx]: do we need add `direction` into the condition also?

File: tests/Table.spec.js [hunk 2]
```diff
@@ -14,6 +14,17 @@ describe('Table.Basic', () => {
 
   describe('renders correctly', () => {
     it('basic', () => {
+      const wrapper = mount(
+        createTable({
+          prefixCls: 'test-prefix',
+          className: 'test-class-name',
+          direction: 'rtl',
```
Comment (zombieJ) [inline tests/Table.spec.js]: Should not change `basic` test with `rtl`. Follow `RTL` is good enough.

Comment (saeedrahimi) [reply tests/Table.spec.js]: That was a mistake.
I wanted to add  `direction: 'rtl'` to `RTL` test case.
.
Fixed.

File: src/hooks/useColumns.tsx (hunk 1 above)
Comment (saeedrahimi) [reply src/hooks/useColumns.tsx]: I made some changes to that class.

**Comment by saeedrahimi** [comment]
@zombieJ I don't know why FixTitle3 is not working even in `LTR` direction. 
..
also tested on `master` branch and still not working.
.
Code coverage fails beacause of this line:
https://codecov.io/gh/react-component/table/pull/426/src/src/hooks/useColumns.tsx#L97
and if its not mandatory i remove children from `revertForRtl` function.

**Comment by zombieJ** [comment]
Excuse me. What is `FixTitle3`?

File: src/hooks/useColumns.tsx [hunk 3]
```diff
@@ -82,6 +82,28 @@ function warningFixed(flattenColumns: { fixed?: FixedType }[]) {
   }
 }
 
+function revertForRtl<RecordType>(columns: ColumnsType<RecordType>): ColumnsType<RecordType> {
+  return columns.map(value => {
```
Comment (zombieJ) [inline src/hooks/useColumns.tsx]: `value`  => `column`

For follow `'children' in column`, it need nest column to test that. You can ref this test: https://github.com/react-component/table/blob/master/tests/GroupingColumns.spec.js

**Comment by saeedrahimi** [comment]
> Excuse me. What is `FixTitle3`?

Capture

Title 3 fixed column doesn't work.

https://github.com/react-component/table/blob/c098e7ac2bafb4a0265a3523c833abcf69ef4a75/examples/fixedColumnsAndHeader.tsx#L51-L57

File: src/hooks/useColumns.tsx (hunk 3 above)
Comment (saeedrahimi) [reply src/hooks/useColumns.tsx]: > `value` => `column`
> 
> For follow `'children' in column`, it need nest column to test that. You can ref this test: https://github.com/react-component/table/blob/master/tests/GroupingColumns.spec.js

changed, and calling `revertForRtl` after flattening columns, so it don't even need children.

https://github.com/react-component/table/blob/da83eada45ea20d2ca7a48ae1b8ed2247c80e77a/src/hooks/useColumns.tsx#L193-L197

**Comment by saeedrahimi** [comment]
All green now, and i think `Fixed Title 3` should be handled in another issue.

**Comment by zombieJ** [comment]
`+ rc-table@7.3.0`

Please help to added in antd@feature brach : )

**Comment by saeedrahimi** [comment]
> `+ rc-table@7.3.0`
> 
> Please help to added in antd@feature brach : )

I'm fixing antd [table bug](https://github.com/ant-design/ant-design/pull/21914) on master branch. , should i move it to feature branch?

**Pull request merged**